src/desert/_make.py

```diff
@@ -370,6 +369,19 @@ def _get_field_default(
         raise TypeError(field)
 
 
+def _get_newtype_supertype(typ: t.Any) -> t.Any:
+    newtype_supertype = getattr(typ, "__supertype__", None)
+    if newtype_supertype:
+        if (
+            # Python<3.10: t.NewType returns a function with a __supertype__ attribute
+            inspect.isfunction(typ)
+            # Python>=3.10: t.NewType is a type
+            or (isinstance(t.NewType, type) and isinstance(typ, t.NewType))
+        ):
+            return newtype_supertype
+    return None
```

Comment:
So I think the complaint about reduced coverage is a false alarm in terms of the change, but probably accurate in terms of us lacking some testing.  You have just exposed it by separating out the conditions into two `if`s which means that the lack of condition coverage before (which we don't measure) has been converted to a lack of branch coverage (which we do measure).  I was able to recreate the complaint on `main`.  I'll see if I can make a quick and good test to cover this now.  I don't want to lose the 100% coverage because it's much easier to track, and I also don't want to pile more existing issues on you.

```patch
diff --git a/src/desert/_make.py b/src/desert/_make.py
index ca37d21..6bd86a3 100644
--- a/src/desert/_make.py
+++ b/src/desert/_make.py
@@ -300,9 +300,10 @@ def field_for_schema(
 
     # t.NewType returns a function with a __supertype__ attribute
     newtype_supertype = getattr(typ, "__supertype__", None)
-    if newtype_supertype and inspect.isfunction(typ):
-        metadata.setdefault("description", typ.__name__)
-        field = field_for_schema(newtype_supertype, default=default)
+    if newtype_supertype:
+        if inspect.isfunction(typ):
+            metadata.setdefault("description", typ.__name__)
+            field = field_for_schema(newtype_supertype, default=default)
 
     # enumerations
     if type(typ) is enum.EnumMeta:
```

```
Name                               Stmts   Miss Branch BrPart     Cover   Missing
---------------------------------------------------------------------------------
src/desert/__init__.py                37      0      0      0   100.00%
src/desert/_make.py                  128      0     72      1    99.50%   304->309
src/desert/_version.py                 1      0      0      0   100.00%
src/desert/exceptions.py               5      0      0      0   100.00%
tests/__init__.py                      0      0      0      0   100.00%
tests/cases/__init__.py                0      0      0      0   100.00%
tests/cases/forward_reference.py      15      0      6      0   100.00%
tests/test_make.py                   287      0     58      0   100.00%
---------------------------------------------------------------------------------
TOTAL                                473      0    136      1    99.84%
```

Reply:
I'm not sure yet why we had the `inspect.isfunction(typ)` check.  It seems like every instance of `NewType` is likely to have a consistent result with that check.

We could also look upstream.

https://github.com/lovasoa/marshmallow_dataclass/blob/65ed180a1a1d5455c14e68d2e562a786ba638e14/marshmallow_dataclass/__init__.py#L704-L707
```python
    # typing.NewType returns a function (in python <= 3.9) or a class (python >= 3.10) with a
    # __supertype__ attribute
    newtype_supertype = getattr(typ, "__supertype__", None)
    if typing_inspect.is_new_type(typ) and newtype_supertype is not None:
```

Reply:
Ha, and `typing_inspect` is already a dependency. Gonna move to it then.